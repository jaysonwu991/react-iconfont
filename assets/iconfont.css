@font-face {
  font-family: "iconfont"; /* Project id 2367065 */
  src: url('iconfont.woff2?t=1664955450032') format('woff2'),
       url('iconfont.woff?t=1664955450032') format('woff'),
       url('iconfont.ttf?t=1664955450032') format('truetype');
}

.iconfont {
  font-family: "iconfont" !important;
  font-size: 16px;
  font-style: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.icon-a-nodejs:before {
  content: "\e715";
}

