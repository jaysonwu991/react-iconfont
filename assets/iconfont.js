window._iconfont_svg_string_2367065='<svg><symbol id="icon-a-nodejs" viewBox="0 0 1024 1024"><path d="M470.25566 12.999716c25.280454-14.470355 58.208227-14.535716 83.466893 0 127.074043 71.803461 254.188028 143.494355 381.240284 215.34139 23.896965 13.460879 39.881532 40.05583 39.638241 67.591263v431.997276c0.177929 28.675631-17.386213 55.906043-42.597674 69.061901-126.656454 71.40766-253.247546 142.902468-379.882213 214.310128-25.803348 14.753589-59.392 13.617021-84.389219-2.345759-37.971518-22.012369-76.008397-43.915801-113.983546-65.906383-7.759887-4.626156-16.507461-8.308199-21.986951-15.832057 4.844028-6.528908 13.504454-7.342298 20.541731-10.192794 15.850213-5.040113 30.407716-13.13044 44.968851-20.981107 3.682043-2.520057 8.177475-1.554156 11.707007 0.700823 32.470241 18.617191 64.653617 37.775433 97.232794 56.214695 6.950128 4.012482 13.987404-1.314496 19.928057-4.626156 124.288908-70.245674 248.730326-140.22627 372.997447-210.49373 4.604369-2.215035 7.149844-7.102638 6.775829-12.146383 0.087149-142.506667 0.021787-285.038752 0.043575-427.545419 0.526525-5.72278-2.785135-10.984397-7.959603-13.326525-126.217078-71.080851-252.368794-142.270638-378.564085-213.369645a13.148596 13.148596 0 0 0-14.866156-0.021787C378.371631 142.550241 252.241702 213.80539 126.046411 284.88261c-5.156312 2.345759-8.616851 7.520227-8.003177 13.264794 0.021787 142.506667 0 285.038752 0 427.567206a12.088284 12.088284 0 0 0 6.862979 11.993872c33.675801 19.096511 67.395177 38.058667 101.092766 57.089816 18.983943 10.218213 42.292652 16.289589 63.208397 8.460709 18.457418-6.619688 31.395404-25.454752 31.043177-45.052369 0.174298-141.675121-0.087149-283.372028 0.130724-425.025361-0.461163-6.289248 5.504908-11.485504 11.619858-10.893617 16.180652-0.108936 32.383092-0.217872 48.563744 0.043574 6.754043-0.152511 11.401986 6.619688 10.566809 12.956142-0.065362 142.572028 0.174298 285.147688-0.108936 427.719716 0.043574 37.996936-15.566979 79.341844-50.713419 97.933617-43.298496 22.429957-96.815206 17.673078-139.590808-3.834553-37.031035-18.482837-72.369929-40.295489-108.743716-60.096454C66.697986 783.926468 49.220993 756.587121 49.398922 727.933277V295.932369c-0.265078-28.105532 16.35495-55.183433 40.996312-68.448227C217.029901 156.018383 343.64278 84.498156 470.25566 12.999716z" fill="#8CC84B" ></path><path d="M580.731461 312.247376c55.230638-3.55495 114.35756-2.106099 164.057872 25.102525 38.479887 20.850383 59.81322 64.610043 60.492256 107.360227-1.074837 5.766355-7.102638 8.947291-12.607546 8.551489-16.024511-0.021787-32.052652 0.217872-48.077164-0.108936-6.797617 0.261447-10.748369-6.006014-11.601702-12.01566-4.604369-20.454582-15.759433-40.713078-35.012085-50.579064-29.554383-14.797163-63.822071-14.052766-96.049021-13.744113-23.526582 1.249135-48.825191 3.286241-68.75688 17.121135-15.301901 10.47966-19.949844 31.635064-14.48851 48.672681 5.14905 12.233532 19.270809 16.180652 30.825305 19.81912 66.56 17.408 137.092539 15.675915 202.381617 38.585192 27.030695 9.339461 53.473135 27.495489 62.725447 55.797106 12.102809 37.931574 6.797617 83.270809-20.189504 113.722099-21.885277 25.05895-53.76 38.697759-85.551206 46.105419-42.292652 9.430241-86.183035 9.669901-129.132936 5.48312-40.38627-4.604369-82.413844-15.214752-113.591376-42.732028-26.660312-23.148936-39.681816-59.217702-38.389106-94.030071 0.308652-5.878922 6.162156-9.978553 11.794156-9.495603 16.137078-0.130723 32.274156-0.174298 48.411234 0.021787 6.44539-0.461163 11.224057 5.109106 11.554497 11.184114 2.977589 19.488681 10.30173 39.943262 27.295773 51.497759 32.79705 21.159035 73.949504 19.710184 111.503432 20.302071 31.11217-1.379858 66.037106-1.797447 91.426497-22.360965 13.395518-11.728794 17.364426-31.35183 13.744113-48.233305-3.921702-14.252482-18.831433-20.893957-31.635064-25.236879-65.706667-20.785021-137.027177-13.243007-202.098383-36.747802-26.417021-9.33583-51.962553-26.987121-62.111773-54.126751-14.161702-38.410894-7.672738-85.921589 22.143092-115.345249 29.071433-29.267518 71.037277-40.53878 110.936965-44.573049z" fill="#8CC84B" ></path></symbol></svg>',function(n){var t=(t=document.getElementsByTagName("script"))[t.length-1],e=t.getAttribute("data-injectcss"),t=t.getAttribute("data-disable-injectsvg");if(!t){var o,i,c,d,a,s=function(t,e){e.parentNode.insertBefore(t,e)};if(e&&!n.__iconfont__svg__cssinject__){n.__iconfont__svg__cssinject__=!0;try{document.write("<style>.svgfont {display: inline-block;width: 1em;height: 1em;fill: currentColor;vertical-align: -0.1em;font-size:16px;}</style>")}catch(t){console&&console.log(t)}}o=function(){var t,e=document.createElement("div");e.innerHTML=n._iconfont_svg_string_2367065,(e=e.getElementsByTagName("svg")[0])&&(e.setAttribute("aria-hidden","true"),e.style.position="absolute",e.style.width=0,e.style.height=0,e.style.overflow="hidden",e=e,(t=document.body).firstChild?s(e,t.firstChild):t.appendChild(e))},document.addEventListener?~["complete","loaded","interactive"].indexOf(document.readyState)?setTimeout(o,0):(i=function(){document.removeEventListener("DOMContentLoaded",i,!1),o()},document.addEventListener("DOMContentLoaded",i,!1)):document.attachEvent&&(c=o,d=n.document,a=!1,r(),d.onreadystatechange=function(){"complete"==d.readyState&&(d.onreadystatechange=null,l())})}function l(){a||(a=!0,c())}function r(){try{d.documentElement.doScroll("left")}catch(t){return void setTimeout(r,50)}l()}}(window);