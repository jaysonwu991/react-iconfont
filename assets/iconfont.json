{
  "id": "2367065",
  "name": "test",
  "font_family": "iconfont",
  "css_prefix_text": "icon-",
  "description": "",
  "glyphs": [
    {
      "icon_id": "32147889",
      "name": "nodejs",
      "font_class": "nodejs",
      "unicode": "e718",
      "unicode_decimal": 59160
    }
  ]
}
