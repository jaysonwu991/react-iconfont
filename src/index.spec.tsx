import React from 'react';
import { render } from '@testing-library/react';

import IconFont from './index';
import '../assets/iconfont.css';

jest.mock('../assets/iconfont.js');

describe('IconFont', () => {
  it('should render successfully when it is icon', () => {
    const { asFragment } = render(<IconFont name='node.js' type='icon' />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render successfully when it is svg symbol', () => {
    const { asFragment } = render(<IconFont name='node.js' type='svg'  />);

    expect(asFragment()).toMatchSnapshot();
  });
});
