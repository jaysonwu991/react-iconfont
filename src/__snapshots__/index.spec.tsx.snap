// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconFont should render successfully when it is icon 1`] = `
<DocumentFragment>
  <span
    style="margin: 0px; padding: 0px;"
  >
    <span
      class="iconfont icon-node.js"
    />
  </span>
</DocumentFragment>
`;

exports[`IconFont should render successfully when it is svg symbol 1`] = `
<DocumentFragment>
  <span
    style="margin: 0px; padding: 0px;"
  >
    <svg
      aria-hidden="true"
      style="width: 1em; height: 1em; vertical-align: -0.15em; fill: currentColor; overflow: hidden;"
    >
      <use
        xlink:href="#icon-node.js"
      />
    </svg>
  </span>
</DocumentFragment>
`;
